nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/SSLContextServiceUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.ssl;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.nifi.components.AllowableValue;
+
+public class SSLContextServiceUtils {
```

Comment:
Does this need to be a separate utility class? It seems to only have one method. I would have anticipated this method being added to the `SSLContextService` interface and then implemented by each of the concrete classes. Thus the logic would be encapsulated in the implementations rather than outside of the provided implementations.

Reply:
@alopresto : I agree and I looked into doing this. However, the problem is that the allowable values for the SSL algorithm PropertyDescriptors are statically initialized with the appropriate values. Adding a method to the `SSLContextService` interface wouldn't enable you to build the values before the class is loaded, which is why i threw a static method into a utility class. This actually reminded me that i forgot to remove the original `buildAllowableValues()` method in `StandardSSLContextService`. I'll do that on my next push.

I can approach this in a different way if you have a recommendation. Thanks!

Reply:
With Java 8, you can provide a `static` method implementation on an `interface`. I'd define a static method `buildAllowableValues()` in `SSLContextService` and implement it with the complete list, and "override" (define a method with the same signature but it's not technically an override in this case) in `RestrictedSSLContextService` which returns the limited list. 

Am I missing something here? I haven't tried it myself but I think this should work.

Reply:
You can indeed have static methods in interfaces, but they cannot be overridden by child classes. They 'belong' to the interface. There were also default methods added in Java 8, which can be overridden by child classes, but they aren't static meaning that they can't be used in statically initialized variables. For these reasons, i elected to put the methods in a util class - but i totally understand the concern here.

See this SO article: https://stackoverflow.com/questions/27833168/difference-between-static-and-default-methods-in-interface

Reply:
Mike, I did read the SO link you had posted but I also read [Java 8 Default and Static Methods](https://www.intertech.com/Blog/java-8-tutorial-default-and-static-methods-guide/) and I think as long as the definitions are on the interfaces, it will be ok.